lib/oli_web/live/workspaces/course_author/history_live.ex

```diff
@@ -0,0 +1,733 @@
+
+  @impl Phoenix.LiveView
+  def handle_info({:updated, revision, _}, socket) do
+    id = revision.id
+
+    revision = Oli.Resources.get_revision!(id) |> Repo.preload(:author)
+
+    revisions =
+      case socket.assigns.revisions do
+        [] -> [revision]
+        [%{id: ^id} | rest] -> [revision] ++ rest
+        list -> [revision] ++ list
+      end
+
+    selected =
+      if revision.id == socket.assigns.selected.id do
+        revision
+      else
+        socket.assigns.selected
+      end
+
+    tree =
+      case Map.get(socket.assigns.tree, revision.id) do
+        nil -> Oli.Versioning.RevisionTree.Tree.build(revisions, socket.assigns.resource_id)
+        node -> Map.put(socket.assigns.tree, revision.id, %{node | revision: revision})
+      end
+
+    {:noreply, assign(socket, selected: selected, revisions: revisions, tree: tree)}
+  end
+
+  @impl Phoenix.LiveView
+  def handle_info({:new_publication, _, _}, socket) do
+    mappings =
+      Publishing.get_all_mappings_for_resource(
+        socket.assigns.resource_id,
+        socket.assigns.project.slug
+      )
+
+    mappings_by_revision =
+      Enum.reduce(mappings, %{}, fn mapping, m -> Map.put(m, mapping.revision_id, mapping) end)
+
+    {:noreply,
+     assign(socket,
+       mappings: mappings_by_revision,
+       publication: determine_most_recent_published(mappings)
+     )}
+  end
+
+  @impl Phoenix.LiveView
+  def handle_params(_params, _url, socket) do
+    {:noreply, socket}
+  end
+
+  @impl Phoenix.LiveView
+  def render(assigns) do
+    ~H"""
+    <div
+      :if={@flash not in [nil, %{}]}
+      id="live_flash_container"
+      class="flash container mx-auto px-0 top-[80px]"
+    >
+      <%= if Phoenix.Flash.get(@flash, :info) do %>
+        <div class="alert alert-info flex flex-row" role="alert">
+          <div class="flex-1">
+            <%= Phoenix.Flash.get(@flash, :info) %>
+          </div>
+
+          <button
+            type="button"
+            class="close"
+            data-bs-dismiss="alert"
+            aria-label="Close"
+            phx-click="lv:clear-flash"
+            phx-value-key="info"
+          >
+            <i class="fa-solid fa-xmark fa-lg"></i>
+          </button>
+        </div>
+      <% end %>
+
+      <%= if Phoenix.Flash.get(@flash, :error) do %>
+        <div class="alert alert-danger flex flex-row" role="alert">
+          <div class="flex-1">
+            <%= Phoenix.Flash.get(@flash, :error) %>
+          </div>
+
+          <button
+            type="button"
+            class="close"
+            data-bs-dismiss="alert"
+            aria-label="Close"
+            phx-click="lv:clear-flash"
+            phx-value-key="error"
+          >
+            <i class="fa-solid fa-xmark fa-lg"></i>
+          </button>
+        </div>
+      <% end %>
+    </div>
+
+    <script
+      id="keep-alive"
+      type="text/javascript"
+      src={Routes.static_path(OliWeb.Endpoint, "/js/keepalive.js")}
+    >
+    </script>
```

Comment:
Do you think this code block is necessary? It is already included in `live.html.heex` which is included in the second line of the module with `use OliWeb, :live_view`. I might be missing something, but I'm concerned that we might be showing flash messages twice in this liveview.

Reply:
Nice catch, yeah sorry I was testing something and forgot to remove it.
Removed in 7e65a8789babca228d4cd9e467004a37997d41e8